packages/core/core/src/services/session-manager.ts

```diff
@@ -197,13 +210,49 @@ class SessionManager {
         sessionId: payload.sessionId,
       };
     } catch (error: any) {
+      // If the token is expired, verify signature ignoring expiration to safely extract payload
+      // and clean up the corresponding session in database.
       if (error instanceof jwt.JsonWebTokenError) {
+        if (error.name === 'TokenExpiredError') {
+          try {
+            const expiredPayload = jwt.verify(token, this.config.jwtSecret, {
+              // Validate signature but ignore exp to retrieve session information
+              ignoreExpiration: true as any,
+            }) as RefreshTokenPayload;
+
+            if (expiredPayload?.sessionId) {
+              await this.provider.deleteBySessionId(expiredPayload.sessionId);
+            }
+          } catch (_) {
+            // If we cannot recover payload safely, skip cleanup
+          }
+        }
         return { isValid: false };
       }
 
       throw error;
     }
   }
+
+  async generateAccessToken(refreshToken: string): Promise<{ token: string } | { error: string }> {
+    const validation = await this.validateRefreshToken(refreshToken);
+
+    if (!validation.isValid) {
+      return { error: 'invalid_refresh_token' };
+    }
+
+    const payload: Omit<AccessTokenPayload, 'iat' | 'exp'> = {
+      userId: validation.userId!,
+      sessionId: validation.sessionId!,
+      type: 'access',
+    };
+
+    const token = jwt.sign(payload, this.config.jwtSecret, {
+      expiresIn: this.config.accessTokenLifespan,
```

Comment:
```suggestion
      algorithm: 'HS256',
      expiresIn: this.config.accessTokenLifespan,
```

Better if we explicitly specify this everywhere (and then easier to find/replace if we go with a different algorithm before release)